donut/modules/arcfeedback/helpers.py

```diff
@@ -0,0 +1,262 @@
+    and updates status of complaint to 'new_msg'
+    if poster is None or an empty string, it will be replaced with
+    "(anonymous)"
+    If complaint_id is invalid, returns False
+    '''
+    if not get_course(complaint_id): return False
+    #add the message
+    query = """
+    INSERT INTO arc_complaint_messages (complaint_id, message, poster, time)
+    VALUES (%s, %s, %s, NOW())
+    """
+    # update the status to new_msg
+    query2 = """
+    UPDATE arc_complaint_info SET status = 'new_msg' WHERE complaint_id = %s
+    """
+    if poster == "" or poster is None:
+        poster = '(anonymous)'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, (complaint_id, message, poster))
+        cursor.execute(query2, complaint_id)
+    return
+
+
+def get_link(complaint_id):
+    """
+    Gets a (fully qualified) link to the view page for this complaint id
+    """
+    query = 'SELECT uuid FROM arc_complaint_info WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchone()
+        if res and 'uuid' in res:
+            uuid = res['uuid']
+        else:
+            uuid = None
+    return flask.url_for(
+        'arcfeedback.arcfeedback_view_complaint', id=uuid,
+        _external=True) if uuid else None
+
+
+def get_id(uuid):
+    """
+    Returns the complaint_id associated with a uuid
+    or false if the uuid is not found
+    """
+    query = 'SELECT complaint_id FROM arc_complaint_info WHERE uuid = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, str(uuid))
+        if not cursor.rowcount:
+            return False
+        res = cursor.fetchone()
+        res = res['complaint_id']
+    return res
+
+
+def get_messages(complaint_id):
+    """
+    Returns timestamps, posters, messages, and message_id's on this complaint
+    in ascending order of timestamp
+    """
+    query = """
+    SELECT time, poster, message, message_id FROM arc_complaint_messages WHERE complaint_id = %s ORDER BY time
+    """
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchall()
+    if not res or 'message_id' not in res[0]:
+        return None
+    return res
+
+
+def get_summary(complaint_id):
+    """
+    Returns a dict with the following fields: course, status
+
+    """
+    fields = ['course', 'status']
+    query = 'SELECT ' + ', '.join(
+        fields) + ' FROM arc_complaint_info WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchone()
+    return res if res else None
+
+
+def get_course(complaint_id):
+    '''
+    Returns the course or None if complaint_id is invalid
+    '''
+    res = get_summary(complaint_id)
+    if res:
+        return res['course']
+    return None
+
+
+def get_status(complaint_id):
```

Comment:
Not saying we should change it for this PR, but we probably want to leave room for an expanded status system depending on how arc wants to organize their feedback. For now it's okay to just have read and unread and have functions specifically for it, but if we wanted more statuses, this would be unsustainable.